TACC/Core-Portal
# WP-38: Prevent multiple clicks to button, add spinner

## Overview
When using the dialog to make a directory (i.e. `DataFilesMkdirModal`):
- user can quickly click twice and create two mkdir operations/requests which are sent to the backend.
- no spinner is shown as the request is being made

## Related
* [FP-38](https://jira.tacc.utexas.edu/browse/FP-38)

## Changes
added isSubmitting function that disables the button when isSubmitting is true. Added spinner to replace button when button is disabled

## Testing
1. In Portal, navigate to Data Files --> My Data(Work)
2. Click +Add [Folder]
3. Enter Name for new Folder
4. Click Create Folder
    At this point, the purple Create Folder button should be a white spinner button that is unclickable. 
5. Repeat steps 2-4 to make sure the button resets to the purple Create Folder. 

## UI
<img width="1440" alt="Screen Shot 2023-06-27 at 2 43 26 PM" src="https://github.com/TACC/Core-Portal/assets/35277477/177d0276-fffd-41b3-84ff-aeb302ab6b4a">

## Notes
We might want to add this fix to other places where the same issue occurs.
